sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -48,6 +49,16 @@ case class Mode(
 
   override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType)
 
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (UnsafeRowUtils.isBinaryStable(child.dataType) || child.dataType.isInstanceOf[StringType]) {
```

Comment:
add a comment here explaining this condition (and why we put it here)

also, adding a TODO: SPARK-### comment is good practice for stuff that we intend to change down the line

Reply:
@uros-db I have added the requested comment. let me know if there is any way to improve what I added.